further_link/runner/process_handler.py

```diff
@@ -68,10 +76,47 @@ async def start(self, command, work_dir=None, env={}):
             process_env["PWD"] = self.work_dir
             process_env["USER"] = self.user
 
-        # Ensure that DISPLAY is set, so that user can open GUI windows
-        display = get_first_display()
-        if display is not None:
-            process_env["DISPLAY"] = display
+        if novnc:
+            self.display = f":{self.id}"
+
+            proc = await asyncio.create_subprocess_shell(
+                f"/usr/bin/further-link-vnc start {self.id}",
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+
+            logging.info(f"Starting further-link-vnc on display '{self.display}'")
+            # Virtual display should be available before executing 'command'
+            await proc.wait()
```

Comment:
we should probably log the command output here, at least as debug level

Reply:
also, i think extract lines 82-90 into a method eg 'start_vnc' in utils/vnc.py

Reply:
maybe even keep starting activity monitor and providing connection details stuff inside the vnc helpers

`await start_vnc(self.id, self.on_display_activity)` - this could cover lines 82-113